### DISCUSSION TITLE
DIY a thermal model

### DISCUSSION THREAD

**Discussion by ImadlakQvQ** [pybamm-team/PyBaMM#3170] [Q&A] [answered]
hi guys:
    now I'm trying to DIY a simplified thermal model based on the x-full thermal model. This DIY model only needs the voltage, current, and Positive open-circuit potential to calculate the total heating. But when I set the RHS of the equation, the domain can't match. The domain of the Positive open-circuit potential is the positive electrode, and the domain of the temperature is the current collector. 
In addition, the shape of the variables cant match either. 
<img width="513" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/65073903/910b4019-6018-4dd9-bbc9-255bf75803e9">

How can I solve this? Can I extract the values of variables and don't consider the domain part? How to do that?
The following is my code:
```
def set_rhs(self, variables):
        T = variables["Cell temperature [K]"]
        I = variables['Current [A]']
        sto = variables['Positive particle surface concentration']/variables['Maximum positive particle concentration']
        Eoc = Uoc(sto)
        E = variables['Local voltage [V]']
        b = pybamm.Parameter('b(x)')
        dEocdT = pybamm.Parameter('dEocdT')
        rho_cp = pybamm.Parameter('rho_cp')
        kx = pybamm.Parameter('kx')
        Q1 = b*I*(-E-T*dEocdT)
        Q2 = b*I*Eoc
        Q = Q2-Q1
        # Fourier's law for heat flux
        q = kx * pybamm.grad(T)
        self.rhs = {T: (pybamm.div(q) + Q) / rho_cp}

```

**Comment by valentinsulzer** [comment, marked as answer]
The domain of the temperature should also be positive electrode.
If you wanted a lumped thermal model, it wouldn't make sense to have d2T/dx2 in there, and also in that case you should use `pybamm.x_average(Eoc)`.
By the way, variables['Positive particle surface concentration'] is already scaled by maximum concentration (min 0, max 1)

**Reply by ImadlakQvQ** [reply]
Thanks for your answering!! 
Sorry for the wrong statement of my question. For more precise understanding of my question, I rewrite the function as following:
<img width="513" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/65073903/05cdf4b0-236a-46e1-a00b-13adc125dc48">
In this way, the function is demontrating a one-dimention thermal model instead of lumped model. 
And I change the Positive open-circuit potential as the built-in variable 'Positive electrode bulk open-circuit potential [V]'.
When I ran the following code, there was a mistake: ' Newton/Linesearch algorithm failed to converge'.
<img width="583" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/65073903/52008332-423b-4b83-b554-ae192d7d584b">
And the values 'q, div(q), ' seem pretty strange.
<img width="1484" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/65073903/acfebe19-2546-49fd-821d-9d9bbaa5cff5">
Is there any mistake I made? Is that ok to use the variable 'Positive electrode bulk open-circuit potential [V]' to calculate the heat ?
And the image shows that the temperature only changes at the edge of the battery, is that normal?
